### DISCUSSION TITLE
Inquiry About Custom Date Rendering in JalaliDatePicker Plugin

### DISCUSSION THREAD

**Discussion by PICLTD** [majidh1/JalaliDatePicker#74] [Question] [answered]
Hello,

I hope you're doing well. I'm currently using the JalaliDatePicker plugin for a project and I have some specific requirements related to date rendering and calendar support that I'm looking to address.

1:	Custom Date Rendering:

I am trying to achieve a specific date rendering behavior that involves enabling only certain days of the week while disabling all others. Specifically, I want to enable only Mondays and Fridays while disabling all other days in the calendar.

I have reviewed the documentation and attempted various configurations, but I haven't been able to achieve the desired result. Here's the code I've been using:

`jalaliDatepicker.startWatch({
  dayRendering: function (dayOptions, input) {
    var daysOfWeek = ["ش", "ی", "د", "س", "چ", "پ", "ج"];
    if (daysOfWeek[dayOptions.dayOfWeek] === "د" || daysOfWeek[dayOptions.dayOfWeek] === "ج") {
      // Enable Monday ("د") and Friday ("ج") by adding a class
      return {
        className: "bg-success text-white"
      };
    } else {
      return {
        isValid: false
      };
    }
  }
});
`

I would greatly appreciate your guidance and assistance in achieving this specific date rendering behavior.

2:	Dual Calendar Support:

Additionally, I would like to offer users the option to switch between the Jalali and Gregorian calendars seamlessly. When a user selects a date, I want the calendar to automatically convert it from Jalali to Gregorian (or vice versa) based on the user's preference.

Could you please provide guidance on how to enable dual calendar support and automatic date conversion within the JalaliDatePicker plugin?

I believe that addressing these requirements would significantly enhance the user experience of the application. Your assistance and insights would be greatly appreciated.

Thank you for your time and support.

**Comment by majidh1** [comment, marked as answer]
Hello,
Thank you for your collaboration. I have created a sample code, and I'm providing you with the link so you can check it out. I think this sample could potentially assist you.

https://codepen.io/majidh1/pen/ExGEZOw

**Reply by PICLTD** [reply]
Thank you, Mr. Majid Hooshiyar, for your invaluable help in solving the issue with the code and for crafting this ingenious solution. Your expertise in customizing the Persian date picker, particularly the thoughtful day rendering and seamless date conversion, is truly commendable. We appreciate the time and effort you dedicated to making this functionality possible!
